### PR TITLE
Suggested changes to `s-flint-defra/issue43`

### PR DESCRIPTION
**Pull request by EFT-Defra**
I'm mainly copying files from `EFT-Defra/issue23` as it's been updated since you created this branch.

### CODE REVIEW

File: .github/workflows/vulnerability-scanning.yml [hunk 1]
```diff
@@ -1,28 +1,27 @@
 # This workflow will run the safety package to check for vulnderabilities in packages if a push includes changes to the poetry.toml or pyproject.lock file.
 
-name: vulnerability_checks
+name: Vulnerability checks
```
Comment (EFT-Defra) [inline .github/workflows/vulnerability-scanning.yml:3]: I don't think you need the underscore here

File: .github/workflows/vulnerability-scanning.yml [hunk 2]
```diff
@@ -1,28 +1,27 @@
...
 on:
   push:
     branches: [ develop ]
     paths:
       - '**/poetry.lock'
       - '**/pyproject.toml'
 jobs:
-  build-and-deploy:
-    concurrency: ci-${{ github.ref }}
+  vulnerability-checks:
```
Comment (EFT-Defra) [inline .github/workflows/vulnerability-scanning.yml:11]: Rename job and remove concurrency, as we're not using it

File: .github/workflows/vulnerability-scanning.yml [hunk 3]
```diff
@@ -1,28 +1,27 @@
...
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8"]
```
Comment (EFT-Defra) [inline .github/workflows/vulnerability-scanning.yml:16]: Add python-version matrix, which is needed for later calls to `${{  matrix.python-version }}`

File: .github/workflows/vulnerability-scanning.yml [hunk 4]
```diff
@@ -1,28 +1,27 @@
...
         with:
           python-version: ${{ matrix.python-version }}
-      - name: Install poetry
-        run: curl -sSL https://install.python-poetry.org | python3 -
-      - name: Install pipx
-        run: python3 -m pip install --user pipx && python3 -m pipx ensurepath
-      - name: Install nox
-        run: pipx install nox && pipx inject nox nox-poetry
+      - name: Setup
+        uses: ./actions/setup-environment
```
Comment (EFT-Defra) [inline .github/workflows/vulnerability-scanning.yml:25]: I consolidated these actions into a single composite action, see `actions/setup-environment/action.yml`

**Comment by EFT-Defra** [comment]
You also need a short write up of what vulnerability scanning is, why we need it, and how to do it. See for example: [documentation-building](https://github.com/Defra-Data-Science-Centre-of-Excellence/sds-data-model/blob/5de5cfd0da4c73cdd6f4c4b04919ccd64cbf16ac/markdown/documentation-building.md),  [formatters](https://github.com/Defra-Data-Science-Centre-of-Excellence/sds-data-model/blob/5de5cfd0da4c73cdd6f4c4b04919ccd64cbf16ac/markdown/formatters.md), or [static-type-analysis](https://github.com/Defra-Data-Science-Centre-of-Excellence/sds-data-model/blob/5de5cfd0da4c73cdd6f4c4b04919ccd64cbf16ac/markdown/static-type-analysis.md). But for expediency, I think that should be a task for a future sprint.

File: .github/workflows/vulnerability-scanning.yml (hunk 2 above)
Comment (s-flint-defra) [reply .github/workflows/vulnerability-scanning.yml:11]: Happy with these changes, if I've understood correctly the common parts of the devtools (set up poetry,nox etc) have been abstracted to the action setup-environment.

**Comment by s-flint-defra** [comment]
Happy with the changes relevant to Issue43, I have only skimmed the changes pulled from Issue23 as outside the scope of this review.

**Pull request merged**